### ISSUE TITLE
statsmodels.stats.power.tt_ind_solve_power able to return non-float types

### ISSUE THREAD

**Issue by vipolsky** [statsmodels/statsmodels#6174]
#### Description

When passed a pandas.Series instead of the expected float value for effect_size, the function will return a pandas.Series. While I realize that my input type is incorrect, I feel the function should either return an error or float type, instead, it will return a pandas.Series.

#### Code Sample

```python
from statsmodels.stats.power import tt_ind_solve_power
import pandas as pd

power = tt_ind_solve_power(effect_size = pd.Series([.8], index = ['a'])
                   , nobs1 = 50
                   , alpha = .95
                   , power = None
                   , ratio = 1.0
                   , alternative = 'two-sided'
                  )

type(power)
```

#### Expected Output

```
ValueError(...
```

#### Actual Output
```
pandas.core.series.Series
```

**Comment by josef-pkt** [comment]
I don't think this was intentional and, I think, it only works if the power or result can be computed directly where vectorized works

However, statsmodels tries in general to return a pandas object if the input is a pandas object

```
power = tt_ind_solve_power(effect_size = pd.Series([.8, 0.05, 0.0001], index = ['a', 'b', 'c'])
                   , nobs1 = 50
                   , alpha = .95
                   , power = None
                   , ratio = 1.0
                   , alternative = 'two-sided'
                  )
```
power is
```
a    1.000000
b    0.951536
c    0.950001
dtype: float64
```

a quick gist

**Comment by josef-pkt** [comment]
There is a mistake, I guess, in your input, alpha should be 0.05
The power was much too high in the example, it looks more reasonable with alpha=0.05.

a variation on the example, where pandas Series is useful

```
nobs1 = np.arange(10, 100, 5)
nobs1_pd = pd.Series(nobs1, index=nobs1)
power = tt_ind_solve_power(effect_size = 0.5
                   , nobs1 = nobs1_pd
                   , alpha = .05
                   , power = None
                   , ratio = 1.0
                   , alternative = 'two-sided'
                  )
power
```

```
10    0.185096
15    0.262443
20    0.337939
25    0.410100
30    0.477897
35    0.540688
40    0.598147
45    0.650186
50    0.696893
55    0.738487
60    0.775266
65    0.807584
70    0.835822
75    0.860368
80    0.881602
85    0.899894
90    0.915587
95    0.929001
dtype: float64
```

However, AFAIK, these are side effects of python duck typing. Most likely there are no unit tests for these cases with vectorized or pandas inputs.

**Comment by josef-pkt** [comment]
I think the behavior here with pandas in then pandas out is fine.

However, there is most likely an inconsistency between cases with explicit formulas and those that need a root finder, numerical root finding is not vectorized, and the return might depend on the scipy.optimize behavior

**Comment by vipolsky** [comment]
Alpha should be 0.05, but this was just an example for the sake of demonstration, so it should not matter. I think better documentation of the pandas in pandas out behavior would be nice. I agree there is nothing fundamentally wrong with this pattern, only that the documentation states the function will return a float and it did not.

**Comment by josef-pkt** [comment]
I agree that the documentation and unit tests should be improved, so we know what actually happens in the different cases.

It's just that the original author (I) never thought about those cases. I think in all examples and test cases we use just plain python scalar numbers (*). I never thought about using pandas for it.

(*) there might be cases with a vectorized numpy array version for specific use cases, e.g. for plotting

**Comment by josef-pkt** [comment]
`type(stats.nct._cdf(50, 2, nobs1_pd))`
`pandas.core.series.Series`

`type(stats.nct.cdf(50, 2, nobs1_pd))`
`numpy.ndarray`

If think this is a change coming from pandas. I'm pretty sure that was not the case in versions from a few years ago.

In the code, I use the underlying `_cdf` method to avoid some generic overhead in the scipy distributions, but this function or method now propagates pandas.
i.e. no np.asarray is used and somehow the computations must be keeping the pandas wrapping as series.
